### PR TITLE
Fix portal url discovery (parsing) in few cases. Prepate to use stb.getUrl for every case. Add alternative expiry check. Prepare editor search to be regex friendly

### PR DESCRIPTION
**Pull request by peshovec**
When adding portal - the url in the form is not yet discovered (parsed). Using it directly to validate mac addresses will fail. Proposed change will allow just to invoke always stb.getUrl - without performance regression, before passing "discovered url" to another function. 
Hope the rest commit messages are clear.

### CODE REVIEW

File: app/templates/editor.html [hunk 1]
```diff
@@ -176,6 +176,10 @@ <h5 class="modal-title" id="channelLabel">Channel</h5>
     /* Create an array with the values of all the text boxes in a column */
     $.fn.dataTable.ext.order['dom-text'] = function (settings, col) {
         return this.api().column(col, { order: 'index' }).nodes().map(function (td, i) {
+            search: {
```
Comment (jojo141185) [inline app/templates/editor.html:179]: Set the search mode inside the map function does not work on my side.
I think it should be set inside the DataTable constructor.

**Comment by jojo141185** [comment]
Is this a duplicate of your other [PR#3](https://github.com/jojo141185/STB-Proxy/pull/3)?

File: app/templates/editor.html (hunk 1 above)
Comment (peshovec) [reply app/templates/editor.html:179]: Good catch:) I overllooked this part. It has to be the same as https://github.com/Chris230291/STB-Proxy/pull/68 just adapted to your code.

**Comment by peshovec** [comment]
Sadly it is not duplicate of the PR#3. I make a mistake. Tried to separate few features/fixes in dedicated PRs and messed them up. 
the "timezone fix" branch (PR#3) seems subset of "explorer" branch (this PR) . 
the explorer branch contains few commits:
 - data file search options - the one you catched as misplaced
 -  fix regex extract - this is functionally important, as it is about discovery of the portal url
 -  stb.getUrl - seems just cosmetic one, but allows to check mac addreses even against url entered in the form (not yet deeplly discovered)
 - the time zone - duplicate with the commit in timezone fix branch

**Comment by jojo141185** [comment]
I merged your PR#3 with timezone feature.
Can you please open a new PR for the other points?

**Comment by peshovec** [comment]
Sure

**Pull request closed without merging**